libtock/timer.h

```diff
@@ -45,14 +45,21 @@ typedef struct tock_timer {
 
 
 /** \brief Create a new alarm to fire in `ms` milliseconds.
+ * 
+ * The `timer` parameter is allocated by the caller and must live as long as
+ * the alarm is outstanding.
+ * 
+ * This should be used for long periods. Due to no reference time, there could
+ * be up to a few milliseconds of inaccuracy. For more accuracy, but less time
+ * range, use `alarm_at`.
```

Comment:
Did the testing show a few milliseconds of inaccuracy? Our timer should have at least millisecond accuracy (we say elsewhere in libtock-c docs that this is an assumption). 

I assume the testing for the accuracy was using `printf` which introduces inherent latency. @ppannuto @bradjc have we benchmarked timers before for accuracy?

Reply:
The testing program has also been updated to print timer error in milliseconds. See my comment to Allistair for a breakdown of our results and how to reproduce the testing: https://github.com/tock/libtock-c/pull/395#discussion_r1569271372

Reply:
`Due to no reference time,`, can we remove that language and replace it with the explanation of precision limitations [once those are more solidly established]?